### PR TITLE
TST: `stats.combine_pvalues`: parameterise tests and update reference values from R

### PR DESCRIPTION
**Pull request by j-bowhay**
https://github.com/scipy/scipy/pull/20900#discussion_r1629664961

Refactor tests to be parameterised and also updates reference values from R

### CODE REVIEW

File: scipy/stats/tests/test_stats.py [hunk 1]
```diff
@@ -7949,50 +7949,41 @@ def test_no_args_gh20661(self):
...
+    # Reference values computed using the following R code:
+    # > options(digits=16)
+    # > library(metap)
+    # > x1 = c(0.01, 0.2, 0.3)
+    # > x2 = c(0.5, 0.5, 0.5)
+    @pytest.mark.parametrize("method,pvalues,expected",
+                             # > sumlog(x1)
+                             [("fisher", [.01, .2, .3], 0.02156175132483465),
+                              # > sumz(x1)
+                              ("stouffer", [.01, .2, .3], 0.01651203260896294),
```
Comment (j-bowhay) [inline scipy/stats/tests/test_stats.py:7962]: updated values pass with default tolerances

File: scipy/stats/tests/test_stats.py [hunk 2]
```diff
@@ -7949,50 +7949,41 @@ def test_no_args_gh20661(self):
...
+                              # > sumz(x2)
+                              ("stouffer", [.5, .5, .5], 0.5),
+                              # need to work out how to generate this
+                              ("pearson", [.01, .2, .3], 0.02213),
```
Comment (j-bowhay) [inline scipy/stats/tests/test_stats.py:7966]: @mdhaber I got stuck finding another library that implements this, any suggestions?

File: scipy/stats/tests/test_stats.py [hunk 3]
```diff
@@ -7949,50 +7949,41 @@ def test_no_args_gh20661(self):
...
-        Z, p = stats.combine_pvalues(xp.asarray([.1, .1, .1]),
-                                     method='mudholkar_george')
-        xp_assert_close(p, xp.asarray(0.019462), rtol=1e-4)
-
-    def test_mudholkar_george_equal_fisher_pearson_average(self, xp):
-        Z, p = stats.combine_pvalues(xp.asarray([.01, .2, .3]),
-                                     method='mudholkar_george')
-        Z_f, p_f = stats.combine_pvalues(xp.asarray([.01, .2, .3]), method='fisher')
-        Z_p, p_p = stats.combine_pvalues(xp.asarray([.01, .2, .3]), method='pearson')
-        xp_assert_close(0.5 * (Z_f+Z_p), Z, rtol=1e-4)
```
Comment (j-bowhay) [inline scipy/stats/tests/test_stats.py:7995]: This test seems redundant, why do we need to test a property if with the same input they are tested against correct reference values

Comment (mdhaber) [reply scipy/stats/tests/test_stats.py:7995]: Could turn this into a property-based test by using random p-values, though.

File: scipy/stats/tests/test_stats.py [hunk 4]
```diff
@@ -7949,50 +7949,40 @@ def test_no_args_gh20661(self):
...
+    # > library(metap)
+    # > x1 = c(0.01, 0.2, 0.3)
+    # > x2 = c(0.5, 0.5, 0.5)
+    @pytest.mark.parametrize("method,pvalues,expected",
+                             # > sumlog(x1)
+                             [("fisher", [.01, .2, .3], 0.02156175132483465),
+                              # > sumz(x1)
+                              ("stouffer", [.01, .2, .3], 0.01651203260896294),
+                              # > sumz(x2)
+                              ("stouffer", [.5, .5, .5], 0.5),
```
Comment (mdhaber) [inline scipy/stats/tests/test_stats.py:7963]: Can we put this in `test_weighted_stouffer` with `weights=None`?

File: scipy/stats/tests/test_stats.py [hunk 5]
```diff
@@ -7949,50 +7949,40 @@ def test_no_args_gh20661(self):
...
-
-    def test_weighted_stouffer(self, xp):
+    # Reference values computed using the following R code:
+    # > options(digits=16)
+    # > library(metap)
+    # > x1 = c(0.01, 0.2, 0.3)
+    # > x2 = c(0.5, 0.5, 0.5)
+    @pytest.mark.parametrize("method,pvalues,expected",
+                             # > sumlog(x1)
+                             [("fisher", [.01, .2, .3], 0.02156175132483465),
```
Comment (mdhaber) [inline scipy/stats/tests/test_stats.py:7959]: Can you update these to test the returned `statistic`, too?

File: scipy/stats/tests/test_stats.py [hunk 6]
```diff
@@ -7949,50 +7949,40 @@ def test_no_args_gh20661(self):
...
+                              # > sumz(x1)
+                              ("stouffer", [.01, .2, .3], 0.01651203260896294),
+                              # > sumz(x2)
+                              ("stouffer", [.5, .5, .5], 0.5),
+                              # > sumlog(x1)
```
Comment (mdhaber) [inline scipy/stats/tests/test_stats.py:7964]: Just a reminder to edit this when you find a reference.

**Comment by mdhaber** [comment]
Remaining comments would be addressed by j-bowhay/scipy#5.

**Comment by mdhaber** [comment]
Thanks @j-bowhay! Would you be willing to add native `axis` support next?

**Comment by j-bowhay** [comment]
> Thanks @j-bowhay! Would you be willing to add native `axis` support next ([#20900 (comment)](https://github.com/scipy/scipy/pull/20900#issuecomment-2154958282))?

Yes, I can do this. Can you remind me, will at need to add further tests or will the generic decorator tests cover this?

**Comment by mdhaber** [comment]
Yeah the generic tests are much stronger than what has typically been written for an individual function, so no need to add anything IMO.

**Pull request merged**